src/svtk/svtk/pesr/sr_test.py

```diff
@@ -9,59 +9,61 @@
-        results['bg_frac'] = results.called / \
-            (results.background + results.called)
-        results['bg_frac'] = results.bg_frac.fillna(0)
-        cols = 'name coord pos log_pval called background bg_frac'.split()
-        return results[cols]
-
     def test_record(self, record, called, background):
         # Test SR support at all coordinates within window of start/end
         resultA = self._test_coord(
-            record, 'posA', record.chrom, called, background)
-        resultA['coord'] = 'posA'
+            record.pos, record.info['STRANDS'][0], record.chrom,
+            called, background, record.pos - self.window, record.pos + self.window)
 
         resultB = self._test_coord(
-            record, 'posB', record.info['CHR2'], called, background)
-        resultB['coord'] = 'posB'
-
-        if (record.chrom == record.info['CHR2']) and (int(resultA.pos) >= int(resultB.pos)):
-            rec = 1
-            while True:
-                result = self._test_coord_V2(
-                    record, 'posB', record.chrom, called, background, rec)
-                if int(result.pos) > int(resultA.pos.iloc[0]):
-                    result['coord'] = 'posB'
-                    resultB = result
-                    break
-                rec = rec + 1
+            record.stop, record.info['STRANDS'][1], record.info['CHR2'],
+            called, background, record.stop - self.window, record.stop + self.window)
+
+        posA = int(resultA.pos)
+        posB = int(resultB.pos)
+        log_pval_A = float(resultA.log_pval)
+        log_pval_B = float(resultB.log_pval)
+        if record.info['SVTYPE'] == 'INS':
+            if posA > posB + self.ins_window or posA < posB - self.ins_window:
+                # Invalid coordinates, need to re-optimize around the better coordinate within the insertion window size
+                if log_pval_A >= log_pval_B:
+                    # posA is better, so use posA as anchor and check for best posB in valid window
+                    resultB = self._test_coord(
+                        posA, record.info['STRANDS'][1], record.info['CHR2'],
+                        called, background, posA - self.ins_window, posA + self.ins_window)
+                else:
+                    # vice versa
+                    resultA = self._test_coord(
+                        posB, record.info['STRANDS'][0], record.chrom,
+                        called, background, posB - self.ins_window, posB + self.ins_window)
+        elif record.chrom == record.info['CHR2'] and posA >= posB:
+            # Invalid coordinates, need to re-optimize around the better coordinate
+            if log_pval_A >= log_pval_B:
+                # posA is better, so use posA as anchor and check for best posB in valid window
+                resultB = self._test_coord(
+                    record.stop, record.info['STRANDS'][1], record.info['CHR2'],
+                    called, background, posA, posA + self.window)
```

Comment:
Does this mean that posA could be returned as the optimal 2nd coordinate because it is in the test window?

Reply:
Yes I think there's a rare corner case here for intrachromosomal variants with `++` and `--` strandedness (INV and some BNDs) that are small (less than twice the window size). In this case with an overlapping window, if the best SR position is within that overlapping window, the algorithm will choose posA=posB, which shouldn't be valid for this variant class. 

Luckily this seems to be a rare case, at least in the 1kgp reference panel there were only 2, and there were no differences in the variants between runs on main and in this branch.

I'm still going to add a check for this case, specifically by rerunning `_test_coord` for posB but explicitly disallowing it to test at posA.